### PR TITLE
move #ifdefs used to enable or disable features from function scope t…

### PR DESCRIPTION
**Pull request by aviateur17**
…o file scope

Takes the #ifdefs used to disable and enable features like filesystem logging, ESP-NOW, LoRa, Debug, WiFi, etc. from function by function to including or excluding entire header files thus being a cleaner or more straightforward and simpler approach.

### CODE REVIEW

File: src/fdrs_gateway_filesystem.h [hunk 1]
```diff
@@ -1,34 +1,23 @@
...
 #endif
 #ifdef USE_FS_LOG
 #include <LittleFS.h>
 #endif
-#if defined(USE_SD_LOG) || defined(USE_FS_LOG)
 #include <time.h>
-#endif
-#if defined(USE_SD_LOG) || defined(USE_FS_LOG)
+
+#define SD_MAX_FILESIZE 1024
```
Comment (aviateur17) [inline src/fdrs_gateway_filesystem.h:11]: SD_MAX_FILESIZE was not previously defined.  Not sure that previously the code would compile without this defined.

File: src/fdrs_gateway_filesystem.h [hunk 2]
```diff
@@ -1,34 +1,23 @@
...
+#define FS_MAX_FILESIZE 1024
```
Comment (aviateur17) [inline src/fdrs_gateway_filesystem.h:12]: FS_MAX_FILESIZE was not previously defined.  Not sure that previously the code would compile without this defined.

File: src/fdrs_gateway_espnow.h [hunk 3]
```diff
@@ -228,86 +223,81 @@ void pingback_espnow()
...
-    }
-    esp_now_send(ESPNOW2, (uint8_t *)&thePacket, j * sizeof(DataReading));
-    esp_now_del_peer(ESPNOW2);
+      esp_now_send(ESPNOW2, (uint8_t *)&thePacket, j * sizeof(DataReading));
+      esp_now_del_peer(ESPNOW2);
 
-    break;
-  }
+      break;
+    }
```
Comment (aviateur17) [inline src/fdrs_gateway_espnow.h:295]: Just added a tab to the previous code for proper identation.

File: src/fdrs_gateway.h [hunk 4]
```diff
@@ -193,3 +215,16 @@ void loopFDRS()
   }
 }
 
+// "Skeleton Functions related to FDRS Actions"
```
Comment (aviateur17) [inline src/fdrs_gateway.h:218]: Skeleton Functions needed to allow these functions to be defined in ACTIONS even when the features are not enabled.   Example:  If LoRa is not enabled that the entire file that has the LoRa functions is excluded from compilation but actions still include broadcastLoRa so need that function present for compilation to succeed.

File: src/fdrs_checkConfig.h [hunk 5]
```diff
@@ -79,8 +79,8 @@ void printLoggingInformation() {
 #else
 	DBG("log buffer delay in ms: NOT SPECIFIED - check config!");
 #endif
-#ifdef FS_FILENAME
-	DBG("log filename          : " + FS_FILENAME);
+#ifdef LOG_FILENAME
```
Comment (aviateur17) [inline src/fdrs_checkConfig.h:82]: I don't believe FS_FILENAME existed in the previous code

File: src/fdrs_globals.h [hunk 6]
```diff
@@ -17,7 +17,8 @@
 // MQTT Topics
 #define TOPIC_DATA    "fdrs/data"
 #define TOPIC_STATUS  "fdrs/status"
-#define TOPIC_COMMAND "fdrs/command" 
+#define TOPIC_COMMAND "fdrs/command"
+#define TOPIC_DATA_BACKLOG "fdrs/databacklog"   // Used in filesystem module
```
Comment (aviateur17) [inline src/fdrs_globals.h:21]: I don't believe TOPIC_DATA_BACKLOG was defined in prior code.

**Comment by aviateur17** [comment]
After modifying the code I went though and made sure all examples compiled using Arduino 1.8.19.  I went through the config files and commented and uncommented the USE_xxx features out and compiled different combinations to make sure all the code works.

**Comment by timmbogner** [comment]
Awesome, thanks! That's gonna make things a lot simpler. I'll get this in before the release, which I'm hoping to do tomorrow.

**Comment by timmbogner** [comment]
ESP8266 does *not* like your ```const``` changes in ESP-NOW.

**Comment by aviateur17** [comment]
Thanks for testing.  I tested against ESP32 board only.   Guess I will have to test ESP8266 in the future.  I've been wondering if there are some automated compile tests that can be done to make sure changes compile against all configurations and all platforms.  I'm not sure if that's called unit testing or what that automated testing would be called.  I gotta believe there is a way to set that up.

**Comment by aviateur17** [comment]
After I finish making these timekeeping changes I'll go back and compile against ESP8266 and update this PR.

**Comment by timmbogner** [comment]
No rush... I'm not going to finish up tonight so I won't get around to pulling it until tomorrow afternoon.

>I gotta believe there is a way to set that up.

Maybe with Arduino CLI and a batch file? That would be really handy. 

Actually... I should look into that in general to streamline my process of editing in the git folder, drag/copy to Arduino folder, click confirm, then clicking compile in Arduino IDE. It's not too bad til you have to fix three small errors in a row and suddenly want to give up 😅

**Comment by aviateur17** [comment]
Fixed issue with const declaration in commit https://github.com/timmbogner/Farm-Data-Relay-System/pull/144/commits/9b5323e447870e7672bccaa8290489dcfc72594d

**Pull request merged**